pelican/contents.py

```diff
@@ -470,9 +486,19 @@ def _log_reason(reason):
 def is_valid_content(content, f):
     try:
         content.check_properties()
-        return True
     except NameError as e:
         logger.error(
             "Skipping %s: could not find information about '%s'",
             f, six.text_type(e))
         return False
+
+    result = True
+    if not content.check_save_as():
+        logger.error(
+            "Skipping %s: file %r would be written outside output path",
+            f,
+            content.save_as,
+        )
+        result = False
+
+    return result
```

Comment:
IMO it looks cleaner to just `return False` inside the `if` block and `return True` at the end, rather than assigning the result status to a variable.